lib/rex/post/meterpreter/client_core.rb

```diff
@@ -617,28 +624,34 @@ def migrate(target_pid, writable_dir = nil, opts = {})
         # Now communicating with the new process
         ###
 
-        # If renegotiation takes longer than a minute, it's a pretty
-        # good bet that migration failed and the remote side is hung.
-        # Since we have the comm_mutex here, we *must* release it to
-        # keep from hanging the packet dispatcher thread, which results
-        # in blocking the entire process.
-        begin
-          Timeout.timeout(timeout) do
-            # Renegotiate SSL over this socket
-            client.swap_sock_ssl_to_plain()
-            client.swap_sock_plain_to_ssl()
+        # only renegotiate SSL if the session had support for it in the
+        # first place!
+        if client.supports_ssl?
+          # If renegotiation takes longer than a minute, it's a pretty
+          # good bet that migration failed and the remote side is hung.
+          # Since we have the comm_mutex here, we *must* release it to
+          # keep from hanging the packet dispatcher thread, which results
+          # in blocking the entire process.
+          begin
+            Timeout.timeout(timeout) do
+              # Renegotiate SSL over this socket
```

Comment:
I wonder if there's a simpler way to do this, since the 'reverse_tcp_ssl' payloads do SSL from the start anyway, even the stager.